Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -12,12 +13,23 @@ public class IonObjectSerializer : IonSerializer<object>
         private readonly IonSerializer ionSerializer;
         private readonly IonSerializationOptions options;
         private readonly Type targetType;
+        private readonly IEnumerable<PropertyInfo> readOnlyProperties;
 
         public IonObjectSerializer(IonSerializer ionSerializer, IonSerializationOptions options, Type targetType)
         {
             this.ionSerializer = ionSerializer;
             this.options = options;
             this.targetType = targetType;
+            
+            if (!this.options.IgnoreReadOnlyProperties && !this.options.IncludeFields)
+            {
+                // If we are serialzing/deserializing readonly properties then we need to serialize/deserialize those
+                // properties' backing fields.
+                // this.readOnlyProperties is used for detecting the appropriate backing fields.
+                // However if we are already including all fields,
+                // then there is no need for this backing field detection logic.
```

Comment:
I don't think the constructor should have this type of logic in it. Rather do this in the Serialize

Reply:
Now this item is lazily instantiated so it only creates if/when we need it.